ale_linters/verilog/iverilog.vim

```diff
@@ -0,0 +1,49 @@
+if exists('g:loaded_ale_linters_verilog_iverilog')
+    finish
+endif
+
+let g:loaded_ale_linters_verilog_iverilog = 1
+
+function! ale_linters#verilog#iverilog#Handle(buffer, lines)
+    " Look for lines like the following.
+    "
+    " tb_me_top.v:37: warning: Instantiating module me_top with dangling input port 1 (rst_n) floating.
+    " tb_me_top.v:17: syntax error
+    " memory_single_port.v:2: syntax error
+    " tb_me_top.v:17: error: Invalid module instantiation
+    let pattern =   '^[^:]\+:\(\d\+\): \(warning\|error\|syntax error\)\(: \(.\+\)\)\?'
+    let output = []
+
+    for line in a:lines
+        let l:match = matchlist(line, pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+
+        let line = l:match[1] + 0
+        let type = l:match[2] ==# 'warning' ? 'W' : 'E'
+        let text = l:match[2] ==# 'syntax error' ? 'syntax error' : l:match[4]
+
+        call add(output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': line,
+        \   'vcol': 0,
+        \   'col': 0,
```

Comment:
I think you'll have to set the column to 1 for some parts of the plugin to work. I'm actually not certain myself. I don't think I've tried it. Maybe this actually works better.